doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,236 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
+
+        .. code-block:: javascript
+
+                var promiseWithError = Promise.resolve().then(function () {
+                    x.y();  // <-- this is an error: x is undefined
+                });
+                promiseWithError.guardedCatch(function (reason) {console.log(reason);}); // will not be called
+                promiseWithError.catch(function (reason) {console.log(reason);}); // will be called
+
+
+
+Testing asynchronous code
+--------------------------
+
+- using promises in tests
+        In the tests code, we support the latest version of Javascript, including primitives like `async` and `await`. This makes using and waiting for promises very easy.
+        Most helper methods also return a promise (either by being marked `async` or by returning a promise directly.
+
+        .. code-block:: javascript
+
+                var testUtils = require('web.test_utils');
+                QUnit.test("My test", async function (assert) {
+                    // making the function async has 2 advantages:
+                    // 1) it always returns a promise so you don't need to define `var done = assert.async()`
+                    // 2) it allows you to use the `await`
+                    assert.expect(1);
+
+                    var form = await testUtils.createView({ ... });
+                    await testUtils.form.clickEdit(form);
+                    await testUtils.form.click('jquery selector');
+                    assert.containsOnce('jquery selector');
+                    form.destroy();
+                });
+
+                QUnit.test("My test - no async - no done", function (assert) {
+                    assert.expect(1);
+
+                    return testUtils.createView({ ... }).then(function (form) {
+                        return testUtils.form.clickEdit(form).then(function () {
+                            return testUtils.form.click('jquery selector').then(function () {
+                                assert.containsOnce('jquery selector');
+                                form.destroy();
+                            });
+                        });
+                    });
+                });
+
+
+                QUnit.test("My test - no async", function (assert) {
+                    assert.expect(1);
+                    var done = assert.async();
+                    testUtils.createView({ ... }).then(function (form) {
+                        testUtils.form.clickEdit(form).then(function () {
+                            testUtils.form.click('jquery selector').then(function () {
+                            assert.containsOnce('jquery selector');
+                            form.destroy();
+                            done();
+                            });
+                        });
+                    });
+                });
+
+        as you can see, the nicer form is to use `async/await` as it is clearer
```

Comment:
I think some explanation are missing between the different examples. Like "If we did not use async, then we would have to force qunit into waiting for the asynchronous end of the test. This is done by chaining all asynchronous code and calling done in the last handler, when we know the test is over." or something like that. Then the end sentence makes more sense IMO: "as you can see, the nice form etc..." since we are now properly introducing the fact that we are comparing them.